tests/e2e/configurer/chain_config.go

```diff
@@ -0,0 +1,21 @@
+package configurer
+
+import "github.com/osmosis-labs/osmosis/v7/tests/e2e/chain"
+
+type ChainConfig struct {
+	chainId         string
+	validatorConfig []*chain.ValidatorConfig
+	// voting period is number of blocks it takes to deposit, 1.2 seconds per validator to vote on the prop, and a buffer.
+	votingPeriod float32
+	// upgrade proposal height for chain.
+	propHeight int
+	// Indexes of the validators to skip from running during initialization.
+	// This is needed for testing functionality like state-sync where we would
+	// like to start a node during tests post-initialization.
+	skipRunValidatorIndexes map[int]struct{}
+	chain                   *chain.Chain
+}
```

Comment:
Is the idea here to move anything calling chain to go through ChainConfig now? If so I will happily change direct references to *chain.Chain if need be

Reply:
Yes, I'm thinking to change the name for `ChainConfig` and better abstract it in a future PR

Reply:
Thinking of either `Network` or `ChainWrapper`. I think the latter is more applicable because it literally just wraps around the `chain,Chain` to have more fields. The ultimate abstraction remains the same. Let me know if you have any preferences